### PR TITLE
Updating README file

### PR DESCRIPTION
**Pull request by paulomarg**
### WHY are these changes introduced?

With all of the updates that we made to this repo, the README file became outdated.

### WHAT is this pull request doing?

Updating the README file to match the current state of the library, and adding some notes on how to migrate from previous versions to the new one.

## Type of change

- [X] Patch: Bug (non-breaking change which fixes an issue)
- [ ] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,16 +15,34 @@ Versions prior to 3.1.68 are vulnerable to a reflected XSS attack. Please update
 
 ## Installation
 
+This package depends on the [Shopify Node Library](https://github.com/Shopify/shopify-node-api) being included in your app, so you should install both of them:
```
Comment (tanema) [inline README.md:18]: *should* install it? Isnt it already installed via dependency?

Comment (paulomarg) [reply README.md:18]: So this is a bit weird, and I haven't had a chance to test this before we release `koa-shopify-auth` - if you use a local import you're not able to import the dependency directly, but I fully expect it to work with an actual package import. I didn't want to state that here until I had a chance to verify it.

File: README.md [hunk 2]
```diff
@@ -83,19 +97,50 @@ app.use(
 );
 ```
 
+### Migrating from cookie-based authentication to session tokens
+
+Versions prior to v4 of this package used cookies to store session information for your app. However, internet browsers have been moving to block 3rd party cookies, which creates issues for embedded apps.
+
+If you have an app using this package, you can migrate to session tokens by performing a few steps:
```
Comment (mkevinosullivan) [inline README.md:104]:
```suggestion
If you have an app using this package, you can migrate from cookie-based authentication to session tokens by performing a few steps:
```

File: README.md [hunk 3]
```diff
@@ -83,19 +97,50 @@ app.use(
...
+
+- Upgrade your `@shopify/koa-shopify-auth` dependency to v4+
+- Install `@shopify/shopify-api` in your backend app
+- Update your server as per the [Usage](#usage) instructions to properly initialize the library and middleware
```
Comment (mkevinosullivan) [inline README.md:108]:
```suggestion
- Update your server as per the [Usage](#usage) instructions to properly initialize the `@shopify/shopify-api` library and the `@shopify/koa-shopify-auth` middleware
```

File: README.md [hunk 4]
```diff
@@ -83,19 +97,50 @@ app.use(
...
+- Install `@shopify/app-bridge-utils` in your frontend app
+- Replace `fetch` calls with `authenticatedFetch` from App Bridge Utils
```
Comment (mkevinosullivan) [inline README.md:110]:
```suggestion
- In your frontend app, replace `fetch` calls with `authenticatedFetch` from App Bridge
```

File: README.md [hunk 5]
```diff
@@ -128,13 +170,9 @@ app
...
 ## Gotchas
 
-### Fetch
-
-This app uses `fetch` to make requests against shopify, and expects you to have it polyfilled. The example app code above includes a call to import it.
-
 ### Session
 
-Though you can use `shopifyAuth` without a session middleware configured, `verifyRequest` expects you to have one. If you don't want to use one and have some other solution to persist your credentials, you'll need to build your own verifiction function.
+The provided `MemorySessionStorage` class is not fit for production use. You can implement your own strategy by creating a class that implements a few key methods. Learn more about [how the Shopify Library handles sessions](https://github.com/Shopify/shopify-node-api/blob/main/docs/issues.md#notes-on-session-handling).
```
Comment (mkevinosullivan) [inline README.md:175]:
```suggestion
The provided `MemorySessionStorage` class may not be scalable for production use. You can implement your own strategy by creating a class that implements a few key methods. Learn more about [how the Shopify Library handles sessions](https://github.com/Shopify/shopify-node-api/blob/main/docs/issues.md#notes-on-session-handling).
```

File: README.md (hunk 2 above)
Comment (paulomarg) [reply README.md:104]: Thanks for the copy check, applying them all!

**Review by mkevinosullivan — APPROVED** [review]

**Pull request merged**